backend/src/api/courses/courses.api.ts

```diff
@@ -17,8 +17,19 @@ const initCoursesApi: FastifyPluginAsync<Options> = async (fastify, opts) => {
   fastify.route({
     method: HttpMethod.GET,
     url: CoursesApiPath.ROOT,
-    async handler(_req, rep) {
-      const courses = await courseService.getAll();
+    async handler(
+      req: FastifyRequest<{ Querystring: { title: string } }>,
+      rep,
+    ) {
+      const { title } = req.query;
+
+      if (!title) {
+        const courses = await courseService.getAll();
+
+        return rep.status(HttpCode.OK).send(courses);
+      }
```

Comment:
It's unnecessary, because you're also checking if filtering opts exists in query
```suggestion

```